packages/flutter_tools/lib/src/android/gradle_errors.dart

```diff
@@ -537,3 +538,25 @@ final GradleHandledError minCompileSdkVersionHandler = GradleHandledError(
   },
   eventLabel: 'min-compile-sdk-version',
 );
+
+@visibleForTesting
+final GradleHandledError jvm11Required = GradleHandledError(
+  test: (String line) {
+    return line.contains('Android Gradle plugin requires Java 11 to run');
+  },
+  handler: ({
+    required String line,
+    required FlutterProject project,
+    required bool usesAndroidX,
+    required bool multidexEnabled,
+  }) async {
+    globals.printBox(
+      '${globals.logger.terminal.warningMark} You need Java 11 or higher to build your app with this version of Gradle.\n\n'
+      'You can update to the latest version of Android Studio on https://developer.android.com/studio/install.\n\n'
```

Comment:
May want to clarify that this also upgrades Java, this is probably not intuitive for new or non-android developers